rust/lance/src/index/vector/ivf/builder.rs

```diff
@@ -67,8 +67,9 @@ pub async fn shuffle_dataset(
             // TODO: Make CPU bound to a future.
             ivf.partition_transform(&batch, column).await
         })
-        .buffer_unordered(num_cpus::get())
-        .and_then(|batch| async move {
+        .buffer_unordered(num_cpus::get() * 2)
+        .map(|batch| async move {
+            let batch = batch.unwrap();
```

Comment:
Is `batch` guaranteed to be ok?  Can we use `and_then` here instead?

Reply:
Sorry, this was sloppy. Fixed.